lib/blockExecution/BlockExecutionContext.js

```diff
@@ -171,6 +173,15 @@ class BlockExecutionContext {
     this.invalidTxs = 0;
     this.consensusLogger = null;
   }
+
+  /**
+   * Clones current instance into separate one
+   *
+   * @returns {BlockExecutionContext}
+   */
+  clone() {
+    return lodashCloneDeep(this);
```

Comment:
I guess we don't need to clone headers and contracts. We can just create a new object and set the same properties